### ISSUE TITLE
Change Request: add a way to get files paths from the eslint api

### ISSUE THREAD

**Issue by pnodet** [eslint/eslint#18877]
### ESLint version

latest

### What problem do you want to solve?

With node.js you can pass a pattern to lintFiles but I didn't find a way to get the list of paths eslint will run on **after** applying configs.

```js
        const pattern = [ "." ] 
        const eslint = new ESLint();
        const results = await eslint.lintFiles(pattern);
```

### What do you think is the correct solution?

Add a method to the ESLint class to return the files depending on a pattern but after applying configs.

```diff
        const pattern = [ "." ] 
        const eslint = new ESLint();
        const results = await eslint.lintFiles(pattern);

+        const lintedFiles = await eslint.getFiles(pattern);
+        console.log("ESLint ran on :" + lintedFiles);
```

### Participation

- [x] I am willing to submit a pull request for this change.

**Comment by pnodet** [comment]
Hey @nzakas sorry, I'd like to use eslint with other tools using the nodejs api, since eslint calculates the paths to run on it would be useful to also pass this information to other tools instead of re-doing the calculations of, for example, ignoring node_modules etc.

**Comment by pnodet** [comment]
I don't mind updating the PR I made and work on it if you have any ideas, I think it would be quite helpful for people building on top of eslint

**Comment by nzakas** [comment]
@pnodet in order to add a method to the public API, we really need a concrete use case to discuss (and also an [RFC](https://github.com/eslint/rfcs).

Why would other tools want to know which files ESLint would lint? What specific other tools are you talking about?

**Comment by pnodet** [comment]
For example running eslint and dprint, disabling conflicting eslint rules and once linting is done via eslint running dprint on the same files. From a unified cli tools.

Same goes for other tools like biome, prettier, etc. Even though I know you can run prettier in eslint you might want to have something that does both separately but avoid having scripts like "lint" : "eslint . && prettier ."

**Comment by pnodet** [comment]
@nzakas Just to make sure we're on the same page this would not modified or impact current project using eslint in node, idea would only be to add a method, not update current workflow

**Comment by nzakas** [comment]
Hmmm, I'm not convinced this is a necessary or useful addition. It sounds like you're trying to micro-optimize a custom toolset, and I'm not sure that's a common enough use case to warrant adding something to the public API. But let's see what the rest of the team thinks.

ping @eslint/eslint-team

**Comment by pnodet** [comment]
I understand, I don't want to push this to much if you don't want to move forward.
I'm in no rush and I can use patch-package or fork if necessary… but wanted to submit the idea here first.
Feel free to close the issue once you've made up your mind!

**Comment by fasttime** [comment]
> For example running eslint and dprint, disabling conflicting eslint rules and once linting is done via eslint running dprint on the same files. From a unified cli tools.

This sounds like a use case for getting a list of files that were _autofixed_ by ESLint, as opposed to any files that ESLint would check if run. Is that information what you are looking for or do you need the list of files without running ESLint?

**Comment by pnodet** [comment]
> This sounds like a use case for getting a list of files that were _autofixed_ by ESLint, as opposed to any files that ESLint would check if run. Is that information what you are looking for or do you need the list of files without running ESLint?

Let's take the example of running `eslint` and `dprint` together.

Getting the files that eslint will run on allow to also run dprint on the same files, even if they didn't report errors or were autofixed I think

**Comment by fasttime** [comment]
> Let's take the example of running `eslint` and `dprint` together.
> 
> Getting the files that eslint will run on allow to also run dprint on the same files, even if they didn't report errors or were autofixed I think

The ESLint Node.js API already reports lint results for all files linted, regardless if they were modified. Again, do you need the list of files without or before running ESLint?

**Comment by nzakas** [comment]
Closing, as we're not going to move forward with this.
